modules/ROOT/pages/indexes-for-nearest-neighbor.adoc

```diff
@@ -0,0 +1,421 @@
+----
+CALL db.index.vector.queryNodes('index_name', numberOfNearestNeighbours, query)
+----
+
+The procedure returns the neighborhood of nodes with their respective similarity scores, ordered by those scores.
+The scores are bounded between `0` and `1`; where the closer to `1`, the more similar to the query vector.
+
+.Query vector index
+====
+For instance, let's take the paper that describes the HNSW{fn-hnsw} graph structure that the vector index implements, and try and find similar papers.
+First we `MATCH` to find the paper, we then query the `abstract-embeddings` index for a neighborhood of `10` similar abstracts to our query.
+Finally, we `MATCH` for the neighborhood's respective titles.
+
+.Query
+[source,cypher]
+----
+MATCH (title:Title)<--(:Paper)-->(abstract:Abstract)
+WHERE toLower(title.text) = 'efficient and robust approximate nearest neighbor search using
+  hierarchical navigable small world graphs'
+
+CALL db.index.vector.queryNodes('abstract-embeddings', 10, abstract.embedding)
+YIELD NODE AS similarAbstract, score
+
+MATCH (similarAbstract)<--(:Paper)-->(similarTitle:Title)
+RETURN similarTitle.text AS title, score
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="5m,2m"]
+|===
+
+| title | score
+
+| "Efficient and robust approximate nearest neighbor search using Hierarchical Navigable Small World graphs"
+| 1.0
+
+| "Accelerating Large-Scale Graph-based Nearest Neighbor Search on a Computational Storage Platform"
+| 0.9437285661697388
+
+| "Nearest Neighbor Search Under Uncertainty"
+| 0.9322342872619629
+
+| "Neighbor selection and hitting probability in small-world graphs"
+| 0.9316230416297913
+
+| "Fast Approximate Nearest Neighbor Search With The Navigating Spreading-out Graph"
+| 0.9314759373664856
+
+| "Towards Similarity Graphs Constructed by Deep Reinforcement Learning"
+| 0.9301378726959229
+
+| "A novel approach to study realistic navigations on networks"
+| 0.928106427192688
+
+| "Intentional Walks on Scale Free Small Worlds"
+| 0.9274556636810303
+
+| "FINGER: Fast Inference for Graph-based Approximate Nearest Neighbor Search"
+| 0.9267876148223877
+
+| "Learning to Route in Similarity Graphs"
+| 0.9263730049133301
+
+2+d| Rows: 10
+
+|===
+
+The results are quite expected, with papers discussing graph-based nearest neighbor searches.
+
+The most similar of the result was the query vector itself, which is to be expected as we queried the index with an indexed property.
+If the query vector itself is not wanted, we can use `WHERE score < 1` to remove equivalent vectors to the query vector.
+
+====
+
+[[indexes-vector-drop]]
+== Drop vector indexes
+
+A vector index is dropped by using the xref:indexes-for-search-performance.adoc#indexes-drop-an-index[same command as for other indexes], `DROP INDEX`.
+
+.+DROP INDEX+
+======
+
+In the following example, we will drop the `abstract-embeddings` that we created previously:
+
+.Query
+[source,cypher]
+----
+DROP INDEX abstract-embeddings
```

Comment:
I'm pretty sure this would need backticks due to the dash, so maybe an unfortunate example name :P